engine/src/main/scala/cromwell/engine/io/IoActor.scala

```diff
@@ -211,7 +216,7 @@ object IoActor {
     case _: SocketException => true
     case _: SocketTimeoutException => true
     case ioE: IOException if Option(ioE.getMessage).exists(_.contains("Error getting access token for service account")) => true
-    case other => isTransient(other)
+    case other => isTransient(other) || isServerError(other)
```

Comment:
@dinvlad if you look at the comment above `isTransient`, i.e. 
```
/**
  * ATTENTION: Transient failures are retried *forever* 
  * Be careful when adding error codes to this method.
  * Currently only 429 (= quota exceeded are considered truly transient)
  */
```
this probably explains why tests carry on forever and time out - since you're adding another way for line 219 to evaluate to `true`, this new way is probably also inducing the infinite retries described.

Reply:
OK, not sure I follow though - calls to `isRetryable` are bounded by `MaxAttemptsNumber`, so they should not be retried "forever" as the comment says. Also, the author didn't modify the logic inside `isTransient` function, so if that was indeed retried indefinitely somehow (if `isTransient` is used elsewhere outside of `isRetryable`), that would've happened previously also..